lighthouse/mcs-compliance.md

```diff
@@ -0,0 +1,176 @@
+# Add support for 100% MCS compliance
+
+Related issue:
+[Add support for 100% MCS Compliance](https://github.com/submariner-io/enhancements/issues/229)
+
+## Summary
+
+Current Lighthouse implementation differs from MCS API [^1] on two key points for ClusterIP Services:
+
+1. There is no VirtualIP [VIP] assigned in `ServiceImport`.
+2. `EndpointSlices` do not contain individual PodIPs.
+
+To better align with MCS API Lighthouse needs an option to allocate and track VirtualIPs in `ServiceImports` and use
+PodIPs in `EndpointSlices`.
+
+## Proposal
+
+When creating `ServiceExport`, users will have option to add annotation for enabling VIP for that Service and another
+to use PodIPs in `EndpointSlices`. These annotations will also be added to the aggregated `ServiceImport`.
+
+Option to add PodIPs to EndpointSlice will only be applicable when VIP option is also enabled. A use case that requires
+PodIPs but not VIP can already be taken care of by using HeadlessServices.
```

Comment:
Then why do we need the EPS option? With VIP enabled, it seems we should just always publish the pod IPs. Wouldn't an external solution need the pod IPs? It seems L51-52 below imply that. I think we should simplify and only have one flag.

Reply:
I was thinking of a scaleable solution where someone wants to use Submariner with VirtualIP and use it to LoadBalance across clusters. e.g. when using something like reference MCS controller.

Reply:
I think we should initially focus on the known use case which is integration with `Istio`, ie VIP + pod IPs. I imagine the use case you mention could come from an upstream user (ie not ACM) and wouldn't be that prevalent, in which case we could add future support via the pod IP annotation only. I'd prefer to minimize the global settings only to the prevalent use case.